src/core/sock/sockinfo.cpp

```diff
@@ -1078,7 +1084,11 @@ bool sockinfo::destroy_nd_resources(const ip_addr &ip_local)
         BULLSEYE_EXCLUDE_BLOCK_START
         unlock_rx_q();
         resource_allocation_key *key;
-        key = m_ring_alloc_logic_rx.get_key();
+        if (m_ring_alloc_logic_rx.get_alloc_logic_type() != RING_LOGIC_PER_IP) {
+            key = m_ring_alloc_logic_rx.get_key();
+        } else {
+            key = m_ring_alloc_logic_rx.create_new_key(ip_local);
```

Comment:
Why do we need to create a key on destroy? isn't it be the same key that was stored on create?